pages/implementation/reference/typescript.mdx

```diff
@@ -0,0 +1,235 @@
+  [systemProperties],
+  [destinationOptions],
+  [customDestination],
+);
+```
+
+The actual parameters depend on your tracking plan setup, see the parameters explanation in <Link passHref href="/implementation/reference/typescript#initavo"><a>the reference below</a></Link>.
+
+#### Step 3. Call Avo functions to track your product usage
+
+Every event in your tracking plan, marked with the "Implement with Avo" checkbox, gets a function in the generated code, named according to the event name, in camelCase.
+
+[//]: # 'TODO Future: Add link here to the "Implement with Avo" doc in "Tracking Plan Management"'
+
+For example, if you have a "Signup Start" event defined like this in Avo:
+
+<center style={{ marginBottom: 16 + 'px' }}>
+  <img
+    style={{ height: 600 + 'px' }}
+    src={require('../../../images/signup-start-ui.png')}
+    alt="Event 'Signup Start' defined in Avo with referral string property and implement with Avo check"
+  />
+</center>
+
+You'll be able to call it like this from the generated code
+
+```typescript
+Avo.signupStart({ referral: 'direct' });
+```
+
+> Notice, that you are not passing the System property with the call. System properties are defined on the init step and then automatically included with all events.
+> You can update the system properties with `setSystemProperties` function.
+
+#### Step 4. Verify the implementation
+
+Use the <Link passHref href="/workspace/tracking-plan/implementation-status"><a>Implementation status</a></Link> in your Avo workspace and the <Link passHref href="/explore-tracking-plan/start-using-visual-debuggers"><a>visual debuggers</a></Link> to verify that your implementation is correct.
+
+### Reference
+
+#### initAvo
+
+```typescript
+Avo.initAvo(
+  options: {
+    env: AvoEnv;
+    [webDebugger?: boolean];
+    [strict?: boolean];
+    [noop?: boolean];
+    [reportFailureAs?: 'error' | 'warn' | 'log'];
+    [inspector?: AvoInspector];
+    [avoLogger?: AvoLogger];
+  },
+  [systemProperties: {
+    systemProperty0: number;
+    systemProperty1: boolean;
+    ...}],
+  [destinationOptions: any],
+  [CustomDestDestination: CustomDestination])
+```
+
+Initializes Avo, needs to be called before the tracking methods. This method will initialize the SDKs for all destinations that are managed by Avo, and call the `make(env)` callback in the provided custom destinations.
+
+##### Arguments
+
+`options`: `{env, [noop], [strict], [avoLogger], [inspector], [mobileDebugger]}`
+
+- `env`: AvoEnv, can be set to dev, prod and staging.
+- `[webDebugger]: optional bool, for optional Avo Web Debugger instance
+- `[strict = false]`: optional bool, if set, Avo will throw an exception when it detects a tracking problem in development or staging. Note that the strict flag is ignored in production.
+- `[noop = false]`: optional bool, if set, Avo won't make any network calls (no tracking) in development and staging environments. Note that the noop flag is ignored in production.
```

Comment:
I think optional is not the correct term here, probably better to call it bool with a default value. Could you also fix it in JS while you are on it?

Reply:
same for strict

Reply:
Like this?

Reply:
Yes, looks very accurate to me now